### PR TITLE
AppThemeBinding on ToolbarItems

### LINKED ISSUE #24878: AppThemeBinding does not work on ToolbarItems
### Description

Hi all,

I didn't find a report that fitted my description, that's why I created a new one. Please feel free to close it as duplicate if I missed the existing report.

In `Styles.xaml` we can set a style for `TargetType="Shell"` and set an `AppThemeBinding` for  `Shell.TitleColor`. As far as I understand this color is used for the page title as well as for the `ToolbarItems` and I expect that the color of the ToolbarItems is by default the same as for the Title. If I now change the theme from light to dark, or dark to light, depending on with which theme you started the app, I would expect that the `ToolbarItem`s change their color according to the theme. Unfortunately, the color remains the same.

Screenshot 2024-09-23 at 17 19 29
Screenshot 2024-09-23 at 17 19 40

I was able to reproduce the behaviour on:
- Android 8
- Android 13
- On iOS 17.6.1 the ToolbarItems don't show up at all.

### Steps to Reproduce

1. Create a new MAUI solution
2. update `Microsoft.Maui.Controls` and `.Compatibility` to 8.0.90
3. Add one or more `ToolbarItem`s to the `MainPage.xaml` or to `AppShell.xaml`
4. Run the app
5. Switch from light to dark theme

Expected:
The ToolbarItem color changes according to the `AppThemeBinding` in style `Shell.TitleColor` and thus have the same color as the title.

Actual:
The ToolbarItem color does not change remains black.

Alternative:
Check out and run my sample project (see below).

### Link to public reproduction project repository

https://github.com/Moes81/maui_toolbaritems_themecolor

### Version with bug

8.0.90 SR9

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 8 and greater, iOS 17.6.1

### Did you find any workaround?

No workaround found so far.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/24878

|Before|After|
|--|--|
|<video src="https://github.com/user-attachments/assets/ead5ebb4-243f-4a13-af07-d43fcecefbd5" width="300px"/>|<video src="https://github.com/user-attachments/assets/efe969c6-bb18-4eb7-83c8-21ef3213ae71" width="300px"/>|

### CODE REVIEW

**Comment by Moes81** [comment]
@kubaflo Thank you for taking the time to fix this. Do you by any chance also tested this with iOS?

**Review by PureWeen — APPROVED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24878.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿#if ANDROID
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24878.cs:1]: Is it relevant to run this test on all platforms?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24878.cs (hunk 1 above)
Comment (kubaflo) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue24878.cs]: I enabled it on Windows

**Review by PureWeen — APPROVED** [review]

**Comment by MauiUIui** [comment]
Does this fix the same issue with Buttons/ImageButtons?

**Pull request merged**